cc/crypto/hpke/recipient_context.cc

```diff
@@ -65,67 +71,43 @@ absl::Status ValidateKeys(std::vector<uint8_t>& public_key_bytes,
 
 absl::StatusOr<std::string> RecipientRequestContext::Open(absl::string_view ciphertext,
                                                           absl::string_view associated_data) {
-  std::vector<uint8_t> ciphertext_bytes(ciphertext.begin(), ciphertext.end());
-  std::vector<uint8_t> associated_data_bytes(associated_data.begin(), associated_data.end());
-
-  // The plaintext should never be longer than the ciphertext.
-  size_t max_out_len = ciphertext_bytes.size();
-  std::vector<uint8_t> plaintext_bytes(max_out_len);
-  size_t plaintext_bytes_len;
-  if (!EVP_HPKE_CTX_open(
-          /* ctx= */ hpke_context_.get(),
-          /* out= */ plaintext_bytes.data(),
-          /* out_len= */ &plaintext_bytes_len,
-          /* max_out_len= */ max_out_len,
-          /* in= */ ciphertext_bytes.data(),
-          /* in_len= */ ciphertext_bytes.size(),
-          /* ad= */ associated_data_bytes.data(),
-          /* ad_len= */ associated_data_bytes.size())) {
-    return absl::AbortedError("Failed to open encrypted message.");
-  }
-  plaintext_bytes.resize(plaintext_bytes_len);
-  std::string plaintext(plaintext_bytes.begin(), plaintext_bytes.end());
+  /// Maximum sequence number which can fit in kAeadNonceSizeBytes bytes.
+  /// <https://www.rfc-editor.org/rfc/rfc9180.html#name-encryption-and-decryption>
+  if (sequence_number_ == UINT64_MAX) {
```

Comment:
This does not match the Rust maximum number.

Reply:
This is pre-existing.

And we are going to get rid of the sequence numbers once we implement random nonces.

Reply:
As context, I used uint64 since C++ does not support uint128 natively (as Rust does).